local-cluster/Main.hs

```diff
@@ -1,127 +1,155 @@
-  (  Options.long "num-wallets"
-  <> Options.long "wallets"
-  <> Options.short 'n'
-  <> Options.metavar "NUM_WALLETS"
-  <> Options.value 1
-  )
+pnumWallets =
+  Options.option
+    Options.auto
+    ( Options.long "num-wallets"
+        <> Options.long "wallets"
+        <> Options.short 'n'
+        <> Options.metavar "NUM_WALLETS"
+        <> Options.value 1
+    )
 
 pdirWallets :: Parser (Maybe FilePath)
-pdirWallets = optional $ Options.strOption
-  (  Options.long "wallets-dir"
-  <> Options.long "wallet-dir"
-  <> Options.short 'd'
-  <> Options.metavar "FILEPATH"
-  )
+pdirWallets =
+  optional $
+    Options.strOption
+      ( Options.long "wallets-dir"
+          <> Options.long "wallet-dir"
+          <> Options.short 'd'
+          <> Options.metavar "FILEPATH"
+      )
 
 padaAmount :: Parser Integer
-padaAmount = Options.option Options.auto
-  (  Options.long "ada"
-  <> Options.short 'a'
-  <> Options.metavar "ADA"
-  <> Options.value 10_000
-  )
+padaAmount =
+  Options.option
+    Options.auto
+    ( Options.long "ada"
+        <> Options.short 'a'
+        <> Options.metavar "ADA"
+        <> Options.value 10_000
+    )
 
 plvlAmount :: Parser Integer
-plvlAmount = Options.option Options.auto
-  (  Options.long "lovelave"
-  <> Options.short 'l'
-  <> Options.metavar "Lovelace"
-  <> Options.value 0
-  )
+plvlAmount =
+  Options.option
+    Options.auto
+    ( Options.long "lovelace"
+        <> Options.short 'l'
+        <> Options.metavar "Lovelace"
+        <> Options.value 0
+    )
 
 pnumUtxos :: Parser Int
-pnumUtxos = Options.option Options.auto
-  (  Options.long "utxos"
-  <> Options.short 'u'
-  <> Options.metavar "NUM_UTXOS"
-  <> Options.value 1
-  )
+pnumUtxos =
+  Options.option
+    Options.auto
+    ( Options.long "utxos"
+        <> Options.short 'u'
+        <> Options.metavar "NUM_UTXOS"
+        <> Options.value 1
+    )
 
 pWorkDir :: Parser (Maybe FilePath)
-pWorkDir = optional $ Options.strOption
-  (  Options.long "working-dir"
-  <> Options.short 'w'
-  <> Options.metavar "FILEPATH"
-  )
+pWorkDir =
+  optional $
+    Options.strOption
+      ( Options.long "working-dir"
+          <> Options.short 'w'
+          <> Options.metavar "FILEPATH"
+      )
 
 pClusterConfig :: Parser CWalletConfig
-pClusterConfig = CWalletConfig
-  <$> pnumWallets
-  <*> pdirWallets
-  <*> padaAmount
-  <*> plvlAmount
-  <*> pnumUtxos
-  <*> pWorkDir
+pClusterConfig =
+  CWalletConfig
+    <$> pnumWallets
+    <*> pdirWallets
+    <*> padaAmount
+    <*> plvlAmount
+    <*> pnumUtxos
+    <*> pWorkDir
 
 -- | Basic info about the cluster, to
 -- be used by the command-line
 data CWalletConfig = CWalletConfig
-  { numWallets :: Int
-  , dirWallets :: Maybe FilePath
-  , adaAmount  :: Integer
-  , lvlAmount  :: Integer
-  , numUtxos :: Int
-  , workDir  :: Maybe FilePath
-  } deriving stock (Show, Eq)
-
+  { numWallets :: Int,
+    dirWallets :: Maybe FilePath,
+    adaAmount :: Integer,
```

Comment:
why in record fields `adaAmount` and `lvlAmount` we are using `Integer` instead of `Positive`, asking this because [here](https://github.com/mlabs-haskell/plutip/blob/fix-local-cluster-args/local-cluster/Main.hs#L61), we convert them to their absolute value anyway. Wouldn't it just be better to don't accept negative amounts, or am I missing something?

Reply:
Good point. `Positive` won't work here as user still need to be able pass `0` amounts, e.g. 
```
cabal run local-cluster -- --wallets 2 --ada 0 --lovelace 888999 --utxos 1 
```
and `Positive` seems like designed to be `> 0`, so I thought `Natural` would fit better here.